design_docs/generalize_calendar.md

```diff
@@ -0,0 +1,304 @@
+days since the reference date 0001-01-01.  
+This is reasonable because the smallest unit of time output in MPAS components is
+seconds (and unlikely to ever be shorter than ms).  We note that a date specified
+as a 64-bit float has a precision high enough to represent seconds for dates up
+to +/- 100 million years: 
+```python
+>>> import sys
+>>> 1./(sys.float_info.epsilon*365*24*60*60)
+142808207.36207813
+```
+We should have no trouble representing any number we might want (including paleo
+timescales) with this system.
+
+For purposes of performing mathematical operations and plotting dates, these
+values will be converted to `datetime.datetime` objects (via the proposed
+`days_to_datetime` utility function) and back (via the proposed 
+`datetime_to_days`).
+
+The conversion operations within `datetime_to_days` and `days_to_datetime` will be
+performed with the calendar-aware functions `netCDF4.date2num` and 
+`netCDF4.num2date`, respectively.  Both functions will support lists/arrays of dates
+(for efficiency and simplicity of calling code) in addition to single values.
+
+Curve ploting can be supported with `matplotlib.pyplot.plot_date`, which takes a date
+of exactly the format used here (days since 0001-01-01), which was part of the reason
+for choosing this format for the date.
+
+<h2> Design solution: The 'Time' coordinate of xarray data sets must support at least years 
+0001 and 9999, and preferably any conceivable value<br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+Same as above. In theory, the use of days since 0001-01-01 would allow any year
+to be supported, not just the range from 0001 to 9999.  However, the conversions
+to `datetime.datetime` objects for mathematical manipulation will constrain
+the dates to be between datetime.min (0001-01-01) and datetime.max (9999-12-31).
+
+
+<h1> Design and Implementation </h1>
+
+<h2> Implementation: The 'Time' coordinate of xarray data sets must be consistent
+with the MPAS calendar <br>
+Date last modified: 2017/02/11 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The proposed implementation is on the branch 
+[xylar/generalize_calendar](https://github.com/xylar/MPAS-Analysis/tree/generalize_calendar)
+
+A helper funciton, `mpas_xarray._parse_dataset_time`, computes times as days since 
+0001-01-01, and serves as a replacement for `mpas_xarray._get_datetimes`. 
+
+**Note: the current implementation breaks the convention that `mpas_xarray` remains
+separate from the rest of MPAS-Analyis by using 3 functions from `timekeeping.utility`
+in `mpas_xarray`:**
+```python
+from ..timekeeping.utility import string_to_days_since_date, \
+    days_to_datetime, datetime_to_days
+```
+**This violates the first requirement in the 
+[MpasXarray class design doc](https://github.com/xylar/MPAS-Analysis/blob/design_doc_variable_mapping_reorg/design_docs/mpas_xarray_class.md).
+I am open to alternative solutions for keeping `mpas_xarray` separate from the rest
```

Comment:
@pwolfram, I'm concerned about this issue and I think you will be, too.  Please let me know what you think and if you have a clean solution.  If a phone call would be best, let me know when you'll be free.